### PR TITLE
Lab25

### PR DESCRIPTION
**Pull request by Alvant**

### CODE REVIEW

**Comment by Alvant** [comment]
Для чего ещё раз разбили на train-test?.. Чтоб со stratify было?
> X_traink1, X_testk1, y_traink1, y_testk1 = train_test_split(...)

**Comment by Alvant** [comment]
> Да, этого не было в задании, но хочу попробовать катбуст

:open_mouth: :+1:

**Comment by Alvant** [comment]
> КНН и ТНН

ТНН?..

**Comment by Alvant** [comment]
> is_normal = np.random.randint(0,2)
> label = "Normal" if is_normal else "Uniform"

? :thinking:

**Comment by Alvant** [comment]
> 0       DBSCAN(eps=1, min_samples=55)

Что-то там не то с ДБСКАНом :sweat_smile: Сколько классов-то он выделил? Они ведь все одним цветом покрашены. Плюс, мне кажется, ДБСКАНу надо было выход TSNE подавать. А то он не может ничего сделать с объектами, у которых сотни признаков. Сложно их сранивать ему, расстояние между ними оценивать.

**Comment by Alvant** [comment]
> Метод понижения размерности НЕ дал выигрыш, возможно дело в гиперпараметрах. Так или иначе, полная размерность дала большую точность

Ну, так, может, не надо было просто так резко уменьшать размерность? Там ведь что-то предлагали до этого поделать, чтоб "эффективную размерность" оценить. Слишком сильное уменьшение числа признаков, выходит, повредило. Много — не очень, мало — не очень. Возможно, "посередине" было бы какое-то хорошее число.

**Comment by Alvant** [comment]
> Здесь Score самый высокий, возможно, дело в том, что этот метод лучше, возможно я нашла лучшие гиперпараметры.

Ну, это не совсем "метод", это про подготовку данных.

**Comment by Alvant** [comment]
> knn.score(new_test_knn.transform(X_test), y_test)

Но KNN ухудшился? Он ведь в самом начале вообще лучший результат показал)

**Comment by Alvant** [comment]
> DBSCAN работает некорректно. Я попробовала различные радиусы и число членов в классе, однако алгоритм всё равно заносит всю выборку в один класс. Жизнь моя жестянка\

Ему надо было давать выборку после уменьшения размерности, после TSNE)

**Comment by AlenaBotay** [comment]
> Для чего ещё раз разбили на train-test?.. Чтоб со stratify было?
> 
> > X_traink1, X_testk1, y_traink1, y_testk1 = train_test_split(...)



> Для чего ещё раз разбили на train-test?.. Чтоб со stratify было?
> 
> > X_traink1, X_testk1, y_traink1, y_testk1 = train_test_split(...)

Да, чтобы попробовать со сбалансированными классами

**Comment by AlenaBotay** [comment]
> > Воооот оно! Катбуст лучше буста
> 
> Но при этом всё равно хуже леса и даже простого KNN?)

Да, в по метрике score knn показал себя лучше всего

**Comment by AlenaBotay** [comment]
> > is_normal = np.random.randint(0,2)
> > label = "Normal" if is_normal else "Uniform"
> 
> ? 🤔

Эта часть была в лабе, я её не трогала
